packages/bierzo-wallet/src/routes/payment/index.tsx

```diff
@@ -1,87 +1,90 @@
+import { ToastContext } from 'medulas-react-components/lib/context/ToastProvider';
+import { ToastVariant } from 'medulas-react-components/lib/context/ToastProvider/Toast';
 import React from 'react';
-import { RouteComponentProps, withRouter } from 'react-router';
+import * as ReactRedux from 'react-redux';
 
-import Header from '../../components/Header';
-import CurrencyToSend from './components/CurrencyToSend';
-import ReceiverAddress from './components/ReceiverAddress';
-import TextNote from './components/TextNote';
+import { history } from '..';
+import { sendSignAndPostRequest } from '../../communication/signAndPost';
+//import { sendSignAndPostRequest } from '../../communication/signAndPost';
+import PageMenu from '../../components/PageMenu';
+import { isIov, lookupRecipientAddressByName } from '../../logic/account';
+import { RootState } from '../../store/reducers';
+import { padAmount, stringToAmount } from '../../utils/balances';
+import { PAYMENT_ROUTE } from '../paths';
+import Layout from './components';
+import { CURRENCY_FIELD, QUANTITY_FIELD } from './components/CurrencyToSend';
+import { ADDRESS_FIELD } from './components/ReceiverAddress';
+import { TEXTNOTE_FIELD } from './components/TextNote';
 
-const useStyles = makeStyles((theme: Theme) => ({
-  payment: {
-    backgroundColor: theme.palette.background.default,
-    gridTemplateColumns: '1fr minmax(375px, 450px) 1fr',
-    gridTemplateAreas: `
-  ". currency-to-send ."
-  ". receiver-address ."
-  ". text-note        ."
-  ". continue-button  ."
-  `,
-    gridGap: '24px',
-    placeItems: 'center',
-  },
+function onCancelPayment(): void {
+  history.push(PAYMENT_ROUTE);
+}
 
-  currencyToSend: {
-    gridArea: 'currency-to-send',
-  },
+const Payment = (): JSX.Element => {
+  const toast = React.useContext(ToastContext);
+  const tokens = ReactRedux.useSelector((state: RootState) => state.tokens);
+  const pubKeys = ReactRedux.useSelector((state: RootState) => state.extension.keys);
 
-  receiverAddress: {
-    gridArea: 'receiver-address',
-  },
+  const onSubmit = async (values: object): Promise<void> => {
+    const formValues = values as FormValues;
 
-  textNote: {
-    gridArea: 'text-note',
-  },
+    const ticker = formValues[CURRENCY_FIELD] as TokenTicker;
+    const amount = stringToAmount(formValues[QUANTITY_FIELD], ticker);
+    const token = tokens[ticker];
+    const chainId = token.chainId;
+    const tokenAmount = padAmount(amount, token.token.fractionalDigits);
 
-  continue: {
-    gridArea: 'continue-button',
-  },
-}));
+    let recipient: Address | undefined;
+    if (isIov(formValues[ADDRESS_FIELD])) {
+      recipient = await lookupRecipientAddressByName(
+        formValues[ADDRESS_FIELD].replace(/\*iov$/, ''),
+        chainId,
+      );
+    } else {
+      recipient = formValues[ADDRESS_FIELD] as Address;
+    }
 
-const onSubmit = (): void => {};
+    if (!recipient) {
+      toast.show('IOV username was not found', ToastVariant.ERROR);
```

Comment:
This error message belong into the `if (isIov(formValues[ADDRESS_FIELD])) {` block. When the user entered a native address, the text makes no sense

Reply:
In case of native address this text will not appear, because `recipient` will not be `undefined`

Reply:
Ah, I see. You can express this fact with code and type system more clearly:

```ts
    let recipient: Address;
    if (isIov(formValues[ADDRESS_FIELD])) {
       const addressForName = await lookupRecipientAddressByName(
        formValues[ADDRESS_FIELD].replace(/\*iov$/, ''),
        chainId,
      );
      if (!addressForName) {
        toast.show('IOV username was not found', ToastVariant.ERROR);
        return;
      }
      recipient = addressForName;
    } else {
      recipient = formValues[ADDRESS_FIELD] as Address;
    }
```

Reply:
To be honest I do not like nested `if`'s :)

Reply:
That's a matter if taste, of course. But nested `if`s allow you to write code that matches the problem that you are facing here. I don't think nested `if`s are a problem as long as you don't nest 5 levels deep